Realm/RLMObject.h

```diff
@@ -709,4 +709,17 @@ typedef void (^RLMObjectChangeBlock)(BOOL deleted,
 @protocol RLM_OBJECT_SUBCLASS <NSObject>   \
 @end
 
+/**
+ Properties on `RLMObject`s of type `RLMSet` must have an associated type. A type is associated
+ with an `RLMSet` property by defining a protocol for the object type that the array should contain.
+ To define the protocol for an object, you can use the macro RLM_SET_TYPE:
+
+     RLM_SET_TYPE(ObjectType)
+     ...
+     @property RLMSet<ObjectType *><ObjectType> *setOfObjectTypes;
+  */
+#define RLM_SET_TYPE(RLM_OBJECT_SUBCLASS)\
```

Comment:
We don't really need a separate macro for this. Maybe it should just be RLM_COLLECTION_TYPE with RLM_ARRAY_TYPE marked as deprecated?

Reply:
I've deprecated `RLM_ARRAY_TYPE ` and updated usage of it across the examples and sdk source.